Use capture phase for document-click-event

Reico3 opened the pull request:
Refactor: use `capture` phase for document click-event.

An event-listener on the on the `document` ensures that the drop-down is closed when the user clicks outside of the `ui-select`.

This behaviour breaks if `event.stopPropagation()` is called on the event before reaching the `document`.

This PR proposes a solution for this problem.

Instead of waiting for a `click` event to bubble up to the document, register the event listener on the capture-phase.

By using capture-phase we guarantee that the callback always is called on the document.

**This relates to the following [Pull-Request](https://github.com/nzzdev/livingdocs-editor/pull/1081). You can find more information there.**

caillou reviewed (approved):
This looks good to me. I updated some of the wording in the description ;)

caillou commented on src/uiSelectDirective.js:
```diff
@@ -207,10 +207,11 @@ uis.directive('uiSelect',
 
         scope.$on('$destroy', function() {
-          $document.off('click', onDocumentClick);
+          $window.document.removeEventListener('click', onDocumentClick);
```
If a listener is registered twice, one with capture and one without, remove each one separately. Removal of a capturing listener does not affect a non-capturing version of the same listener, and vice versa.

So you need to add the third `true` parameters.

Reico3 committed:
test: testing if dropdown closes even when propagation of event stops

Reico3 committed:
fix(drop-down): close drop-down regardless of `stopPropagation()` calls

As of now, an event-handler on the `document` is responsible for closing the
drop-down. A click-event stops bubbling when `stopPropagation()` is called,
therefor it breaks the closing of the drop-down.

Registering the event-listener in capture phase fixes this bug.
